libbeat/processors/ratelimit/rate_limit.go

```diff
@@ -120,6 +126,13 @@ func (p *rateLimit) makeKey(event *beat.Event) (uint64, error) {
 	return hashstructure.Hash(values, nil)
 }
 
+func (p *rateLimit) tagEvent(event *beat.Event) {
+	if p.config.MetricField != "" && p.numRateLimited.Load() > 0 {
+		event.PutValue(p.config.MetricField, p.numRateLimited.Load())
+		p.numRateLimited.Store(0)
```

Comment:
I am on the fence about whether to reset this counter each time or let it grow so it's a true counter. I decided to not go with the latter since it would be more susceptible to wraparound issues. But happy to discuss and change!